pysat/utils/_core.py

```diff
@@ -570,6 +579,147 @@ def generate_instrument_list(inst_loc, user_info=None):
+        inst_info = dict()
+
+        # Cycle through the available instrument modules
+        for inst_mod in inst_mods:
+            # Get the platform, name, and instrument module name
+            platform = inst_mod.split('_')[0]
+            name = '_'.join(inst_mod.split('_')[1:])
+            mod_name = '.'.join([inst_loc.__name__, inst_mod])
+
+            # Initialize the dictionary for this platform and name
+            if platform not in inst_info.keys():
+                inst_info[platform] = dict()
+            inst_info[platform][name] = {'inst_module': mod_name}
+
+            # Get the inst_id and tag information, if possible
+            try:
+                module = importlib.import_module(mod_name)
+                inst_info[platform][name]['inst_ids_tags'] = {
+                    inst_id: {tag: module.tags[tag]
+                              for tag in module.inst_ids[inst_id]}
+                    for inst_id in module.inst_ids.keys()}
+            except ImportError as ierr:
+                inst_info[platform][name]['inst_ids_tags'] = {
+                    'ERROR': {'ERROR': str(ierr)}}
+
+    return inst_info
+
+
+def display_available_instruments(inst_loc=None, show_inst_mod=None,
+                                  show_platform_name=None):
+    """Display basic information about instruments in a given subpackage.
+
+    Parameters
+    ----------
+    inst_loc : python subpackage or NoneType
+        The location of the instrument subpackage (e.g., pysat.instruments)
+        or None to list all registered instruments (default=None)
+    show_inst_mod : boolean or NoneType
+        Displays the instrument module if True, does not include it if False,
+        and reverts to standard display based on inst_loc type if None.
+       (default=None)
+    show_platform_name : boolean or NoneType
+        Displays the platform and name if True, does not include it if False,
+        and reverts to standard display based on inst_loc type if None.
+       (default=None)
+
+    Note
+    ----
+    Prints to standard out, a user-friendly interface for availabe_instruments.
+    Defaults to including the instrument module and not the platform/name values
+    if inst_loc is an instrument module and to including the platform/name
+    values and not the instrument module if inst_loc is None (listing the
+    registered instruments).
+
+    """
+
+    inst_info = available_instruments(inst_loc)
+
+    if show_platform_name is None and inst_loc is None:
+        show_platform_name = True
+
+    if show_inst_mod is None and inst_loc is not None:
+        show_inst_mod = True
+
+    if show_platform_name:
+        header = "Platform   Name   "
+    else:
+        header = ""
+
+    if show_inst_mod:
+        header = "{:s}Instrument_Module".format(header)
+
+    header = "{:s}      [Tag   Inst_ID]  Description".format(header)
+    print(header)
+    print("-" * len(header))
+    for platform in inst_info.keys():
```

Comment:
I looked at the registry and since it is a dict order isn't guaranteed. Adding a sort here would make it easier for a user to parse a long list.
```suggestion
    for platform in np.sort(inst_info.keys()):
```
This simple suggestion only sorts by platform name, which may be weird for a listing by instrument module.

Reply:
I added a sort that happens only if platform is being displayed.